docs/how-to/simple-authpolicy-for-app-developers.md

```diff
@@ -0,0 +1,254 @@
+
+## Overview
+
+Authenticated rate limiting rate limits the traffic directed to an application based on attributes of the client user, who is authenticated by some authentication method. A few examples of authenticated rate limiting use cases are:
+- User A can send up to 50rps ("requests per second"), while User B can send up to 100rps.
+- Each user can send up to 20rpm ("request per minute").
+- Admin users (members of the 'admin' group) can send up to 100rps, while regular users (non-admins) can send up to 20rpm and no more than 5rps.
+
+In this guide, we will rate limit a sample REST API called **Toy Store**. In reality, this API is just an echo service that echoes back to the user whatever attributes it gets in the request. The API exposes an endpoint at `GET http://api.toystore.com/toy`, to mimic an operation of reading toy records.
+
+We will define 2 users of the API, which can send requests to the API at different rates, based on their user IDs. The authentication method used is **API key**.
+
+| User ID | Rate limit                             |
+|---------|----------------------------------------|
+| alice   | 5rp10s ("5 requests every 10 seconds") |
+| bob     | 2rp10s ("2 requests every 10 seconds") |
+
+
+### ①  Deploy the Toy Store API
+
+Create the deployment:
+
+```sh
+kubectl apply -f - <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: toystore
+  labels:
+    app: toystore
+spec:
+  selector:
+    matchLabels:
+      app: toystore
+  template:
+    metadata:
+      labels:
+        app: toystore
+    spec:
+      containers:
+        - name: toystore
+          image: quay.io/3scale/authorino:echo-api
+          env:
+            - name: PORT
+              value: "3000"
+          ports:
+            - containerPort: 3000
+              name: http
+  replicas: 1
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: toystore
+spec:
+  selector:
+    app: toystore
+  ports:
+    - name: http
+      port: 80
+      protocol: TCP
+      targetPort: 3000
+EOF
+```
+
+Create a HTTPRoute to route traffic to the service via Istio Ingress Gateway:
+
+![](https://i.imgur.com/rdN8lo3.png)
+
+```sh
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: toystore
+spec:
+  parentRefs:
+    - kind: Gateway
+      name: prod-web
+      namespace: kuadrant-multi-cluster-gateways
+  hostnames:
+  - toystore.$MGC_ZONE_ROOT_DOMAIN
+  rules:
+  - matches:
+    - path:
+        type: Exact
+        value: "/toy"
+      method: GET
+    backendRefs:
+    - name: toystore
+      port: 80
+EOF
+```
+
+Verify the route works:
+
+```sh
+curl -ik https://toystore.$MGC_ZONE_ROOT_DOMAIN/toys
+# HTTP/1.1 200 OK
+```
+
+### ② Enforce authentication on requests to the Toy Store API
+
+Create a Kuadrant `AuthPolicy` to configure the authentication:
+
+```sh
+kubectl apply -f - <<EOF
+apiVersion: kuadrant.io/v1beta1
```

Comment:
I guess we're merging this based on v1beta1 first and then reiterate once v1beta2 is out.

Reply:
Yes have re-worked this a bit for v1beta1 instead, for now. Will revisit again when v1beta2 is nearly out.